rbi/core/hash.rbi

```diff
@@ -362,6 +362,22 @@ class Hash < Object
   sig {returns(T::Enumerator[[K, V]])}
   def delete_if(&blk); end
 
+  # Retrieves the value object corresponding to the each key objects repeatedly.
+  #
+  # ~~~ruby
+  # h = { foo: {bar: {baz: 1}}}
+  #
+  # h.dig(:foo, :bar, :baz)           #=> 1
+  # h.dig(:foo, :zot)                 #=> nil
+  # ~~~
+  sig do
+    params(
+      key: K
+    )
```

Comment:
Digging into the errors on the mono-repo, they're arising from the nested hashes not having the same key type as the outer-most hash. I think we'll need to have the key argument be `T.untyped`, or do something like:
```
sig {params(key: K, rest: T.untyped).returns(T.untyped)
def dig(key, *rest); end
```